src/Databases/DatabaseReplicatedWorker.cpp

```diff
@@ -77,6 +77,39 @@ String DatabaseReplicatedDDLWorker::enqueueQuery(DDLLogEntry & entry)
     return enqueueQueryImpl(zookeeper, entry, database);
 }
 
+
+bool DatabaseReplicatedDDLWorker::waitForReplicaToProcessAllEntries(UInt64 timeout_ms)
+{
+    auto zookeeper = getAndSetZooKeeper();
+    UInt32 our_log_ptr = parse<UInt32>(zookeeper->get(database->replica_path + "/log_ptr"));
+    UInt32 max_log_ptr = parse<UInt32>(zookeeper->get(database->zookeeper_path + "/max_log_ptr"));
+    assert(our_log_ptr <= max_log_ptr);
+
+    /// max_log_ptr is the number of the last successfully executed request on the initiator
+    /// The log could contain other entries which are not committed yet
+    /// This equality is enough to say that current replicas is up-to-date
+    if (our_log_ptr == max_log_ptr)
+        return true;
+
+    /// We create an empty entry to a the log to reuse existing approach (a bit of copy-paste)
+    /// One big disadvantage is that it triggers other replicas to update their state which is not really needed
+    DDLLogEntry entry{};
+    auto entry_path = enqueueQueryImpl(current_zookeeper, entry, database, /*committed=*/true);
+    auto entry_name = entry_path.substr(entry_path.rfind('/') + 1);
```

Comment:
So we cannot simply use `wait_current_task_change` condvar, because it's triggered when replica starts to process the task and not when it finishes, right? Can we use a watch on `/log_ptr` node?

Reply:
Do not really understand. We currently use the `wait_current_task_change` condvar and I though that additional triggering and "false-positive" signals won't we a problem, but now I think they will. Think using another condvar with the same mutex will be enough.

Reply:
And how to use a watch? Because it will triggered each time `/log_ptr` updated.. We will have to check whether it equal to /max_log_ptr and if not - resubscribe.

Reply:
Yep, we have to resubscribe each time the watch triggered. Another option is to use `wait_current_task_change` to wait for replica to start processing `max_log_ptr` entry and use watch to wait for it to finish.